### DISCUSSION TITLE
I'm encountering an error message 'Reached maximum retries (3) for URL ...' when using the HTTP request component in Dify.

### DISCUSSION THREAD

**Discussion by zhengjianping1010** [langgenius/dify#10322] [General]
### Self Checks

- [X] I have searched for existing issues [search for existing issues](https://github.com/langgenius/dify/issues), including closed ones.
- [X] I confirm that I am using English to submit this report (我已阅读并同意 [Language Policy](https://github.com/langgenius/dify/issues/1542)).
- [X] [FOR CHINESE USERS] 请务必使用英文提交 Issue，否则会被关闭。谢谢！:）
- [X] Please do not modify this template :) and fill in all the required fields.

### Content

In my dify app,I called other dify app, but, 'Reached maximum retries (3) for URL ...' was displayed.
Why?

**Comment by zhengjianping1010** [comment]
@dosubot

**Reply by RRRoger** [reply]
how to fix it, my dify version is 0.12.1

**Comment by Hello-Pig** [comment]
使用 HTTP请求 节点 以及 使用自定义工具中的http请求，都有这个问题！
折腾了半天，还以为是接口问题，，，，
怎么都调用不通，报错信息也没有，很迷

**Reply by Hello-Pig** [reply]
补充下：之前老的 自定义工具 还是可以在流程中使用的，但是测试也是不通，很奇怪。就是之前的服务有依赖之前的自定义工具，没有问题，单独测试就是不通

以及 新建的 自定义工具 HTTP 请求也不通

**Reply by Hello-Pig** [reply]
现在的 HTTP请求节点 和 自定义工具 处于不可用状态

**Reply by zhengjianping1010** [reply]
感谢回答。
源码没有仔细看，估计这样用可能会引起比较奇怪的问题，所以现在可能给屏蔽掉了。
我主要想要用户信息传进来，看了下源码，web访问发送的request中，不包含user信息，
没办法了，只能在外面包一层，然后通过HTTP访问的方式，访问Dify的API了。

**Reply by gracechen104** [reply]
I installed it with docker and removed SSRF. There was no problem accessing public APIs, but accessing domain-restricted APIs was not possible. I used the curl command in the docker container to confirm that the network environment could access the API.

**Reply by gubinjie** [reply]
SSRF使用HTTPX发送请求，HTTPX默认超时时间5s，新版本增加了HTTPX的超时时间设置

**Reply by akira0912** [reply]
> SSRF使用HTTPX发送请求，HTTPX默认超时时间5s，新版本增加了HTTPX的超时时间设置

Does this mean that I can solve this problem by upgrading the DIFY version?

**Reply by gubinjie** [reply]
@akira0912 The timeout is already maintained with the local deployment, but with the dockerCompose deployment, it seems that no one is maintaining it
